tools/configure_server_settings.rb

```diff
@@ -4,19 +4,26 @@
 
 opts = Trollop.options(ARGV) do
   banner "USAGE:   #{__FILE__} -s <server id> -p <settings path separated by a /> -v <new value>\n" \
-         "Example: #{__FILE__} -d -s 1 -p reporting/history/keep_reports -v 42"
+         "Example: #{__FILE__} -d -s 1 -p reporting/history/keep_reports -v 42\n" \
+         "Example: #{__FILE__} -s 1 -p workers/worker_base/queue_worker_base/ems_metrics_collector_worker/defaults/count -v 4 -i"
 
   opt :dry_run,  "Dry Run",                            :short => "d"
-  opt :serverid, "Server Id",                          :short => "s", :default => 0
-  opt :path,     "Path within advanced settings hash", :short => "p", :default => ""
-  opt :value,    "New value for setting",              :short => "v", :default => ""
+  opt :serverid, "Server Id",                          :short => "s", :type => :integer, :required => true
+  opt :path,     "Path within advanced settings hash", :short => "p", :type => :string,  :required => true
+  opt :value,    "New Value for setting",              :short => "v", :type => :string,  :required => true
+  opt :integer,  "Value Provided is an Integer",       :short => "i", :type => :boolean, :default => false
```

Comment:
I didn't go this route in #17043 because it's not obvious it's required for some values...

Reply:
See below @rvalente, I think warning the user they're changing an existing Integer value to a String and telling them to look at the help banner to specify an integer may make this option easier to use.

Reply:
Agreed, I'll improve the output

Reply:
@rvalente Was looking at #17043 with @jrafanie and we were thinking that this approach would be better if, instead of having the boolean option `:integer`  
```
opt :integer,  "Value Provided is an Integer",       :short => "i", :type => :boolean, :default => false
```
an option that specifies the type of the value, defaulting to `string` and only accepts `string`, `integer`, `boolean`, `symbol`  and `float`.  Something like this -
```
TYPES = %w{ string integer boolean symbol float }
opt :type,  "Type of value provided, #{TYPES.inspect}",       :short => "t",  :type => :string, :default => "string"

Trollop.die :type, "must be one of #{TYPES.inspect}" unless TYPES.include?(opts[:type])
```

I think it would still be good to warn if the new type of the new value is different than the old one.

Reply:
I agree, this make sense to me.  In this way, we're not forcing casting of strings to other types.

We default to string and use the check mentioned below to detect when we're changing a sensible default from the correct type to a String type (count from 1,2,3, etc. to "1", "2", "3") and request the user to either allow this to happen or inform them they need to specify the type.

Reply:
Note, integer was requested here, but boolean was also requested [here](https://github.com/ManageIQ/manageiq/pull/16992) so having a single flag that allows us to specify a type would help solve all of these at once.